orchestrator/ethereum_gravity/src/utils.rs

```diff
@@ -167,8 +169,36 @@ pub async fn get_event_nonce(
     Ok(downcast_uint256(real_num).expect("EventNonce nonce overflow! Bridge Halt!"))
 }
 
+/// Gets gravity id, on failure: if `wait_time` is present it will retry until elapsed
+/// if not, it will keep retrying indefinitely.
+pub async fn get_gravity_id_with_retry(
```

Comment:
What is the purpose of getting it indefinitely? I would stick to retiring but with a limited number of attempts or at least operation time (btw we already have this values)